utils/index.d.ts

```diff
@@ -2100,3 +2123,239 @@ export declare namespace validationUtils {
+         * can choose an arbitrary subscription for the agent to use as a default value for the "pick a subscription" prompt. This allows
+         * the agent to move onto more important prompts like the "choose a storage account type" prompt.
+         */
+        useAsSkipValue?: boolean;
+    };
+} & T;
+
+export type AgentQuickPickOptions<T extends IAzureQuickPickOptions = IAzureQuickPickOptions> = { agentMetadata: ParameterAgentMetadata; } & T;
+
+export type AgentInputBoxOptions<T extends AzExtInputBoxOptions = AzExtInputBoxOptions> = {
+    agentMetadata: ParameterAgentMetadata & {
+        /**
+         * A value that the agent can use as a sort of "default" value in order to skip answering the input box prompt this options object is
+         * associated with. This is useful for input boxes that don't have any dependents, as the agent can avoid getting stuck trying to answer
+         * them. Once the user chooses to go with the parameters that the agent has picked, they will be asked to input a value for the input box
+         * prompt this options object is associated with.
+         */
+        skipValue?: string;
+    }
+} & T;
+
+/**
+ * An interface compatible with {@link IAzureUserInput} that allows for the use of an agent to answer prompts instead of showing prompts to the user. Wizards/wizard steps
+ * for commands that are exposed to an agent should use this interface to make sure that in the case of an agent being the one to answer prompts, that all necessary
+ * information is provided to the agent in order to answer the prompts.
+ */
+export interface IAzureAgentInput {
+    readonly onDidFinishPrompt: Event<PromptResult>;
+    showQuickPick<ItemsBaseT extends QuickPickItem, OptionsBaseT extends IAzureQuickPickOptions>(items: AgentQuickPickItem<ItemsBaseT>[] | Promise<AgentQuickPickItem<ItemsBaseT>[]>, options: AgentQuickPickOptions<OptionsBaseT> & { canPickMany: true }): Promise<AgentQuickPickItem<ItemsBaseT>[]>;
+    showQuickPick<ItemsBaseT extends QuickPickItem, OptionsBaseT extends IAzureQuickPickOptions>(items: AgentQuickPickItem<ItemsBaseT>[] | Promise<AgentQuickPickItem<ItemsBaseT>[]>, options: AgentQuickPickOptions<OptionsBaseT>): Promise<AgentQuickPickItem<ItemsBaseT>>;
+    showInputBox<OptionsBaseT extends IAzureQuickPickOptions>(options: AgentInputBoxOptions<OptionsBaseT>): Promise<string>;
+
+    showWarningMessage<T extends MessageItem>(message: string, ...items: T[]): Promise<T>;
+    showWarningMessage<T extends MessageItem>(message: string, options: IAzureMessageOptions, ...items: T[]): Promise<T>;
+    showOpenDialog(options: AzExtOpenDialogOptions): Promise<Uri[]>;
+}
+
+// #endregion
+
+// #region Agent integration types
+
+/**
+ * A config that describes a command that the extension implements which makes use of wizards that use
+ * an {@link IAzureAgentInput}/{@link IAzureUserInput} to get user input.
+ */
+export type WizardBasedCommandConfig = {
+    type: "wizard";
```

Comment:
What's with this? Will this include other types in the future?

Reply:
I hope so, and I'd like to leave this here as part of that hope.